src/utils/validationUtil.ts

```diff
@@ -0,0 +1,229 @@
+import { isDate } from "lodash";
+import * as enums from "../assets/Enums";
+import * as models from "../models/";
+
+
+export class ValidationUtil {
+
+  public static checkTestResults(data: models.ITestResultData) {
+    if (data) {
+      if (!data.Count) {
+        throw new models.HTTPError(404, enums.ERRORS.NoResourceMatch);
+      }
+    }
+    return data.Items;
+  }
+
+  public static validateGetTestResultFilters(filters: models.ITestResultFilters) {
+    if (!(filters && ValidationUtil.validateDates(filters.fromDateTime, filters.toDateTime))) {
+      console.log("Invalid Filter");
+      return false;
+    }
+    return true;
+  }
+
+  private static validateDates(fromDateTime: string | number | Date, toDateTime: string | number | Date) {
+    return fromDateTime !== undefined && toDateTime !== undefined && isDate(new Date(fromDateTime)) && isDate(new Date(toDateTime)) && isFinite((new Date(fromDateTime)).getTime()) && isFinite((new Date(toDateTime)).getTime());
+  }
+
+  public static isMissingRequiredCertificateNumberOnAdr(
+    payload: models.ITestResultPayload
+  ): boolean {
+    return ValidationUtil.isMissingRequiredCertificateNumber(
+      ValidationUtil.isTestTypeAdr,
+      payload
+    );
+  }
+
+  public static isMissingRequiredCertificateNumberOnTir(
+    payload: models.ITestResultPayload
+  ): boolean {
+    return ValidationUtil.isMissingRequiredCertificateNumber(
+      ValidationUtil.isTestTypeTir,
+      payload
+    );
+  }
+
+  public static isMissingRequiredCertificateNumber(
+    typeFunc: (testType: models.TestType) => boolean,
+    payload: models.ITestResultPayload
+  ): boolean {
+    let bool = false;
+    if (payload.testTypes) {
```

Comment:
something consider so we avoid having this temporary bool variable is to cast the result with !!() however I am not sure whether we have instances where we have booleans represented as strings as if it is the case 'false' with cause issues as it will be coerced to true...